seisbench/models/base.py

```diff
@@ -192,6 +193,113 @@ def download_metadata_callback(metadata_path):
+            if file.endswith("json"):
+                with open(path / file, "r") as json_file:
+                    weights_metadata = json.load(json_file)
+            elif file.endswith("pt"):
+                model_weights = torch.load(f"{path / file}")
+
+        def _get_args(obj):
+            signature = inspect.signature(obj)
+            args = {k: v._default for k, v in signature.parameters.items()}
+            if "kwargs" in args.keys():
+                del args["kwargs"]
+            return args
+
+        model_args = weights_metadata.get("model_args", {})
+
+        # Overwrite default class args from config model args
+        cls_args = _get_args(cls)
+        cls_args.update({k: model_args[k] for k in cls_args if k in model_args})
+
+        model = cls(**cls_args)
+
+        model._weights_metadata = weights_metadata
+        model._parse_metadata()
+
+        model.load_state_dict(model_weights)
+
+        return model
+
+    def save(self, path=None, name=None):
+        """
+        Save a SeisBench model locally.
+
+        SeisBench models are stored inside the directory 'path'. SeisBench models are saved in 2 parts,
+        the model configuration is stored in JSON format [path]/[name.json], and the underlying model weights
+        in PyTorch format [path]/[name].pt.
+
+        The model config should contain the following information, which is automatically created from
+        the model instance state:
+            - "docstring": A string documenting the pipeline. Usually also contains information on the author.
+            - "model_args": Argument dictionary passed to the init function of the pipeline.
+            - "seisbench_requirement": The minimal version of SeisBench required to use the weights file.
+            - "default_args": Default args for the :py:func:`annotate`/:py:func:`classify` functions.
+
+        Non-serlisable arguments (e.g. functions) cannot be saved to JSON, so are not converted.
+
+        :param path: Define the output model directory. If none is provided, the path will create a directory
+                     to store the the model in the current working directory, the name of this directory will
+                     be the class name in lowercase, defaults to None
+        :type path: pathlib.Path or None, optional
+        :param name: Define the name of the model. If none is provided, the model name will be automatically
+                     inferred as the class name in lowercase, defaults to None
+        :type name: str, optional
+        """
+        if not name:
+            name = self.name.lower()
+
+        if not path:
+            path = Path(f"{name}")
+        path.mkdir(parents=True, exist_ok=True)
+
+        model_args = {
+            k: v
+            for k, v in self._argspec.items()
+            if k not in ("__class__", "self", "default_args") and not callable(v)
+        }
+        model_metadata = {
+            "docstring": self.__class__.__doc__,
```

Comment:
I think using `__class__.__doc__` is not the optimal solution. I'd rather add the docstring as a parameter to the save function. Maybe it could be empty by default but seisbench returns a warning if you actually keep it empty?

Reply:
Just on this, what's the reasoning behind dropping the docstring? My thinking was that as we are always saving / loading the config to reproduce some child '_model_' class, inheriting from SeisBench model, the docstring of this child class will stay the same? Is there a case I'm missing where we would want a different docstring?

Reply:
The `docstring` argument should describe the weights, not the model. The model is already described in the class/documentation. Saving a property of the class in the json file would be redundant as the json can anyhow only be loaded if you have the implementation of the class available.

Reply:
Ah ok got you! Yeah, I was thinking this was a bit of circular argument when writing, but that was all coming from my misunderstanding of 'docstring' - I can update this now, cheers!